core/json/src/main/scala/net/liftweb/json/JsonAST.scala

```diff
@@ -874,20 +843,70 @@ object JsonAST {
     }
   }
 
+  sealed trait AppendContainer[T] {
```

Comment:
I'm going to try and make some time to try again myself this week and see if I observe the same differences (unless you beat me to it)—since Scala's `StringBuilder` uses Java's behind the scenes, it seems more likely that the difference in performance was a fluke, and this has added enough abstraction weirdness that it'd be ideal if we could avoid it.